### ISSUE TITLE
use wired solution with Wizen module on teensy

### ISSUE THREAD

**Issue by stephanschulz** [gilmaimon/ArduinoWebsockets#52]
Your library looks great. It seems to be more updated than 
https://github.com/PaulStoffregen/ArduinoWebsocketClient

I am hoping to use Websocket through the wizen module on a teensy 3.2 (Arduino compatible micro controller).
But the library I am currently using does not support SEC-Key but it looks yours does. 

Do you think it would be possible to add support for the wiznet ethernet module? 
Your wifi solution is great but I find a wired solution more secure for long term installations.

Thanks for any advice.

**Comment by adelin-mcbsoft** [comment]
Yeah, WebSockets for wired Ethernet Shield (W5100, WizNet W5500 or ENC28j60) for MCUs like ATMega2560, AMtel328p or Cortex M0 would be a great idea!
We could donate some boards if needed for research & testing.

**Comment by gilmaimon** [comment]
Hi guys, 

As you know, currently the library only supports esp32 and esp8266. Adding support for other boards is not very complicated, but it does take time, which I don't have plenty of.

I can however describe the process and let you try adding support for it yourselvs, If it works for you, you can consider donating it to the library. The process is:
1. Implement the interfaces `TcpClient` and `TcpServer` for your board. [Here](https://github.com/gilmaimon/ArduinoWebsockets/blob/master/src/tiny_websockets/network/esp32/esp32_tcp.hpp) is an example implementation of esp32 client and server (client uses [this](https://github.com/gilmaimon/ArduinoWebsockets/blob/master/src/tiny_websockets/network/generic_esp/generic_esp_clients.hpp) generic esp code)
2.  Register it in [ws_common.hpp](https://github.com/gilmaimon/ArduinoWebsockets/blob/master/src/tiny_websockets/internals/ws_common.hpp). Meaning: add a `#elif defined` for your board.

This is the process generally, some experimination is required for sure.

**Comment by stephanschulz** [comment]
Thanks for those tips.
I will certainly give it a try and keep everyone updated here.
